Sources/SwiftDiscord/Rest/DiscordRateLimiter.swift

```diff
@@ -152,30 +152,65 @@ public final class DiscordRateLimiter {
 /// Ex. /channels/232184444340011009/messages and /channels/186926276592795659/messages
 /// Are considered different endpoints
 public struct DiscordRateLimitKey: Hashable {
+	/// URL Parts for the purpose of rate limiting.
+    /// Combine all the parts of the URL into a list of which parts exist
+    /// Ex. /channels/232184444340011009/messages would be represented by [.channels, .channelID, .messages]
+    public struct DiscordRateLimitURLParts: OptionSet {
```

Comment:
Just so I understand correctly. This basically just creates an option for each url component. But doesn't that limit the number of possible endpoints this could represent?